modules/common/src/main/java/org/opencastproject/mediapackage/elementbuilder/AbstractElementBuilderPlugin.java

```diff
@@ -143,15 +143,9 @@ else if (!file.isFile())
       throw new IllegalStateException("Mime types have not been initialized");
 
     // Check mimetype
-    if (mimeTypes != null) {
-      if (mimeTypes.size() > 0) {
-        for (MimeType m : mimeTypes) {
-          if (m.equals(MimeTypes.fromURL(file.toURI().toURL())))
-            return true;
-        }
-      } else {
-        return false;
-      }
+    for (MimeType m : mimeTypes) {
```

Comment:
I would rather recommend to replace all code of this method by 

```
  if (mimeTypes != null) {
    return checkMimeType(file, mimeTypes);
  } else {
    throw new IllegalStateException("Mime types have not been initialized");
  } 

```

;-)

Reply:
@lkiesow Actually, it seems like the object instance variable "mimeTypes" is never initialised... does this method work at all?